docs/api-irma-server.md

```diff
@@ -147,3 +162,102 @@ Also returns a session result JWT, but one whose structure is the same as the se
+
+Note that all endpoints below use the `/irma/session/{clientToken}/...` prefix. This prefix exactly
+matches the `u` field from the `sessionPtr` object, so the location of `/irma` endpoints can also be derived doing
+`{sessionPtr.u}/...`. A `sessionPtr` can be obtained using the [`POST /session`](#post-session) requestor endpoint.
+
+---
+
+### `DELETE /irma/session/{clientToken}`
+Behaves exactly the same as the [delete endpoint for requestors](#delete-session-token), but uses the
+client token from the `sessionPtr` instead of the requestor token.
+
+---
+
+### `GET /irma/session/{clientToken}/status`
+Behaves exactly the same as the [status endpoint for requestors](#get-session-token-status), but uses the client token
+from the `sessionPtr` instead of the requestor token.
+
+---
+
+### `GET /irma/session/{clientToken}/statusevents`
+Behaves exactly the same as the [statusevents endpoint for requestors](#get-session-token-statusevents), but uses the
+client token from the `sessionPtr` instead of the requestor token.
+
+---
+
+### `POST /irma/session/{clientToken}/frontend/options`
+This endpoint can be used by [frontend libraries](irma-frontend.md) to set options for an IRMA session to
+make it match the used frontend flow. A request for specific options can be sent to this endpoint and
+in the response an overview of the active session options is sent. This includes the changes made by
+processing the options request. If an options request holds an invalid combination of options, an error
+is returned.
+
+Session options can be changed multiple times. However, as soon as an
+[irmaclient](https://github.com/privacybydesign/irmago/tree/master/irmaclient)/[IRMA app](irma-app.md)
+has connected to the session, it is not possible to change the options anymore. In other words, this
+endpoint can only be used when the [session status](#get-session-token-status) is `"INITIALIZED"`.
+
+To make sure this endpoint can only be accessed by frontends, requests must be done with an
+additional `Authorization` HTTP request header. The expected value for this request header
+is the `frontendAuth` token, as received along with the `sessionPtr` from
+the [`POST /session`](#post-session) requestor endpoint.
+
+The body of an options request should have the following structure:
+```json
+{
+  "@context": "https://irma.app/ld/request/options/v1",
+  "pairingMethod": "..."
+}
+```
+
+Currently we only have one option, the option `pairingMethod`. It can have two values:
+ * `"pairingMethod": "none"` **(default value)**  
+   No device pairing is used.
+ * `"pairingMethod": "pin"`  
+   When an [irmaclient](https://github.com/privacybydesign/irmago/tree/master/irmaclient)/[IRMA app](irma-app.md)
+   connects to the session, first a `PAIRING` [session status](#get-session-token-status) is introduced 
+   between the `INITIALIZED` and `CONNECTED` status. Both the irmaclient and the frontend show a 4 digit pairing code and
```

Comment:
The introduction of that status occured when you programmed it, not during each session. Suggestion:
```
When an [irmaclient](https://github.com/privacybydesign/irmago/tree/master/irmaclient)/[IRMA app](irma-app.md)
connects to a session in which pairing is enabled, the [session status](#get-session-token-status) becomes `PAIRING`.
```
Then later:
```
... only after the user confirms in the frontend that both PINs match the session continues,
and the status becomes `CONNECTED`.
```

By the way, doesn't the latest/current version of the shoulder-surfing branch of `irma-frontend` ask the user to type that PIN, instead of displaying it?

Reply:
Yeah true, I didn't update that yet here, so that needs to be done.